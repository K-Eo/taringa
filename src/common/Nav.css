.Nav {
  background-color: #005dab;
  border-bottom: 1px solid #004178;
}
