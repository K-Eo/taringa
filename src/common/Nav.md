Basic Nav

```js
const { MemoryRouter } = require('react-router-dom');

<MemoryRouter>
  <Nav />
</MemoryRouter>
```
