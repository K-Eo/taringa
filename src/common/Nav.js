import React from 'react'
import { Link, NavLink } from 'react-router-dom'

import './Nav.css'
import SearchInput from '../search/components/SearchInput'

const Nav = props => {
  const isActive = (match, { pathname }) => {
    return pathname === '/' || pathname === '/recents'
  }

  return (
    <div className="navbar navbar-expand-lg navbar-dark Nav">
      <div className="container">
        <Link className="navbar-brand font-weight-bold" to="/">
          TARINGA!
        </Link>

        <div className="collapse navbar-collapse">
          <div className="form-inline">
            <SearchInput />
          </div>

          <div className="navbar-nav ml-5 mr-auto">
            <NavLink
              activeClassName="active"
              className="nav-item nav-link font-weight-bold"
              isActive={isActive}
              to="/"
            >
              MI T!
            </NavLink>

            <NavLink
              activeClassName="active"
              className="nav-item nav-link font-weight-bold"
              to="/global"
            >
              GLOBAL
            </NavLink>

            <NavLink
              activeClassName="active"
              className="nav-item nav-link font-weight-bold"
              to="/channels"
            >
              CANALES
            </NavLink>
          </div>

          <div className="navbar-nav">
            <a className="nav-item nav-link" href="/">
              Iniciar sesi√≥n
            </a>
            <a className="nav-item nav-link" href="/">
              Crear cuenta
            </a>
          </div>
        </div>
      </div>
    </div>
  )
}

export default Nav
