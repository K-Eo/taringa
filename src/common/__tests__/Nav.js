import React from 'react'
import renderer from 'react-test-renderer'

import Nav from '../Nav'

describe('Nav', () => {
  it('renders default', () => {
    const tree = renderer.create(<Nav />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
