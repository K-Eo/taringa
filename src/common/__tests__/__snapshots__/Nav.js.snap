// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders default 1`] = `
<div
  className="navbar navbar-expand-lg navbar-dark Nav"
>
  <div
    className="container"
  >
    <Link
      className="navbar-brand font-weight-bold"
      replace={false}
      to="/"
    >
      TARINGA!
    </Link>
    <div
      className="collapse navbar-collapse"
    >
      <div
        className="form-inline"
      >
        <withRouter(Connect(SearchInput)) />
      </div>
      <div
        className="navbar-nav ml-5 mr-auto"
      >
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav-item nav-link font-weight-bold"
          isActive={[Function]}
          to="/"
        >
          MI T!
        </NavLink>
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav-item nav-link font-weight-bold"
          to="/global"
        >
          GLOBAL
        </NavLink>
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav-item nav-link font-weight-bold"
          to="/channels"
        >
          CANALES
        </NavLink>
      </div>
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link"
          href="/"
        >
          Iniciar sesi√≥n
        </a>
        <a
          className="nav-item nav-link"
          href="/"
        >
          Crear cuenta
        </a>
      </div>
    </div>
  </div>
</div>
`;
